build/config-content-safari.mjs

```diff
@@ -1,9 +1,68 @@
 import VueLoaderPlugin from '../node_modules/vue-loader/lib/plugin.js'
 import MiniCssExtractPlugin from 'mini-css-extract-plugin'
 import path from 'path'
+import fs from 'fs'
 const projectRoot = process.cwd()
 
+/**
+ * This plugin injects a version and a build name into Safari plist files.
+ */
+class plistPlugin {
+  /**
+   *
+   * @param {object} buildInfo - An object that contains the following build information:
+   * @param {string} buildInfo.branch - A name of a git branch that a code being built belongs to.
+   * @param {string} buildInfo.number - An string in the format of YYYYMMDDCCC, where
+   *         YYYY is a year, MM is a month, DD is a day, and CCC is a build timestamp.
+   * @param {string} buildInfo.name - A build name in the branch-name.YYYYMMDDCCC format.
+   */
+  constructor ({ buildInfo } = {}) {
+    this.buildInfo = buildInfo
+  }
+
+  apply (compiler) {
+    // The code below will run bu the `done` webpack hook
+    compiler.hooks.done.tapPromise('PlistPlugin', async (stats) => {
+      const projectRoot = process.cwd()
+      const pkgPath = path.join(projectRoot, 'package.json')
```

Comment:
we already define this variable on the start of the file, but I suggest even to move all the calss to a separate file  in the other comment

Reply:
Agree, will do that